.github/workflows/e2e_integration_test.yml

```diff
@@ -19,24 +19,84 @@ env:
   CHANNEL_ACCOUNT_ENCRYPTION_PASSPHRASE: ${{ vars.DISTRIBUTION_SEED }}
   SEP10_SIGNING_PUBLIC_KEY: ${{ vars.SEP10_SIGNING_PUBLIC_KEY }}
   SEP10_SIGNING_PRIVATE_KEY: ${{ vars.SEP10_SIGNING_PRIVATE_KEY }}
+  CIRCLE_API_KEY: ${{ vars.CIRCLE_API_KEY }}
+  CIRCLE_USDC_WALLET_ID: ${{ vars.CIRCLE_USDC_WALLET_ID }}
+
 jobs:
-  e2e-integration-test-stellar:
-    runs-on: ubuntu-latest
-    environment: "Receiver Registration - E2E Integration Tests (Stellar)"
-    env:
-      DISTRIBUTION_ACCOUNT_TYPE: "DISTRIBUTION_ACCOUNT.STELLAR.ENV"
-    steps:
-      - uses: actions/checkout@v4
-      - id: stellar
-        uses: ./.github/workflows/actions/e2e_integration_test
-  e2e-integration-test-circle:
+  e2e:
     runs-on: ubuntu-latest
-    environment: "Receiver Registration - E2E Integration Tests (Circle)"
+    strategy:
+      matrix:
+        platform:
+          - "Stellar"
+          - "Circle"
+        include:
+          - platform: "Stellar"
+            environment: "Receiver Registration - E2E Integration Tests (Stellar)"
+            DISTRIBUTION_ACCOUNT_TYPE: "DISTRIBUTION_ACCOUNT.STELLAR.ENV"
+          - platform: "Circle"
+            environment: "Receiver Registration - E2E Integration Tests (Circle)"
+            DISTRIBUTION_ACCOUNT_TYPE: "DISTRIBUTION_ACCOUNT.CIRCLE.DB_VAULT"
```

Comment:
Used the `matrix`as a form to re-execute the same tests multiple times and still keep the steps easily debuggable, as well as the Docker logs.

Reply:
nice!